code/__HELPERS/~monkestation-helpers/traits.dm

```diff
@@ -0,0 +1,15 @@
+/datum
+	/// Lazylist of trait cascade targets. Don't modify this directly, refer to 'cascade_trait()' instead.
+	var/list/_cascade_targets
+
+/// Links 'trait' to 'target_trait', such that when 'trait' is added/removed 'target_trait' is too. Uses 'trait' as the source.
+/datum/proc/cascade_trait(trait, target_trait)
+	RegisterSignal(src, SIGNAL_ADDTRAIT(trait), PROC_REF(_cascade_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(trait), PROC_REF(_cascade_trait_loss))
```

Comment:
you should set these to `override = TRUE` in case this is called multiple times, or just return if the cascade trait is already set

Reply:
It should runtime if its registered twice because that means someone fucked up

Reply:
I.e. used cascade twice (bad) or used it in combination with signal addtrait (bad)

Reply:
Cascade is init signals only, i should update the docs a bit

Reply:
oh fair